src/drivers/wdm/queries/ExaminedValue/ExaminedValue.ql

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/**
+ * @name ExaminedValue
+ * @kind problem
+ * @description The returned value is annotated with the _Check_return_ or _Must_inspect_result_ annotation, but the calling function is either not using the value or is overwriting the value without examining it. For more information please refer C28193 Code Analysis rule.
+ * @problem.severity warning
+ * @id cpp/portedqueries/examined-value
+ * @platform Desktop
+ * @feature.area Multiple
+ * @repro.text The following code locations potentially contain function calls whose return values are not checked.
+ * @version 1.0
+ */
+
+import cpp
+import drivers.wdm.libraries.WdmDrivers
+import drivers.libraries.SAL
+
+//Represents functions that are annotated with either _Check_return_ or _Must_inspect_result_
+class ReturnMustBeCheckedFunction extends Function {
+  SALCheckReturn scr;
+
+  ReturnMustBeCheckedFunction() { this.getADeclarationEntry() = scr.getDeclarationEntry() }
+}
+
+class ReturnMustBeCheckedFunctionCall extends FunctionCall {
+  ReturnMustBeCheckedFunctionCall() { this.getTarget() instanceof ReturnMustBeCheckedFunction }
+}
+
+//Holds if an expression (a call to ReturnMustBeCheckedFunction in this case) is occuring in a void context.
+predicate unUsed(Expr e) {
+  e instanceof ExprInVoidContext
+  or
+  definition(_, e.getParent()) and
+  not definitionUsePair(_, e.getParent(), _)
+}
+
+/**
+ * In the general case this predicate is used to calculate the number of checked & total calls to a ReturnMustBeCheckedFunction. In the other case (if the two lines below are uncommented), it evaluates to true for ReturnMustBeCheckedFunctionCall's whose return values have been used in at aleast "X" % of the total number of calls.
+ */
+predicate callFrequency(ReturnMustBeCheckedFunction f, string message) {

```

Comment:
How does the percentage of checks help the end user identify/fix their bugs? Would it be more helpful to provide the names of the functions which are not always checked? Or perhaps the file and line numbers of where return values are used but not checked?

Reply:
The percentage gives users additional information in how the function is being used. For example, if there are 10 total calls and 9 of them are checking the return value before proceeding, then the user will see a message that says '90% of the total number of calls are checking the return value' and that will help the user review/rethink their code and potentially avoid subsequent failures. If on the other hand only 10% of the total number of calls are checking the return value, then the user may not pay that much attention to the defect even though it's a true defect. 

The function names and their line numbers are outputted when the codeql analysis command is run.